src/Uno.UI/UI/Xaml/Controls/Image/ImageSizeHelper.cs

```diff
@@ -1,122 +1,179 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Text;
 using Windows.Foundation;
 using Windows.UI.Xaml.Media;
+using Uno.UI;
 using static Windows.UI.Xaml.Media.Stretch;
 
 namespace Windows.UI.Xaml.Controls
 {
 	internal static class ImageSizeHelper
 	{
-		public static void MeasureSource(this Image image, Windows.Foundation.Rect parent, ref Windows.Foundation.Rect child)
+		public static void MeasureSource(this Image image, Size finalSize, ref Rect child)
 		{
 			switch (image.Stretch)
 			{
 				case UniformToFill:
-					var uniformToFillScale = (child.Width * parent.Height >= child.Height * parent.Width)
-						? parent.Height / child.Height // child is flatter than parent
-						: parent.Width / child.Width; // child is taller than parent
+					var uniformToFillScale = (child.Width * finalSize.Height >= child.Height * finalSize.Width)
```

Comment:
Do we validated upfront that `child.Height` and `child.Width` ?

Reply:
What do you mean? That's the size of the image pixels. If you're here, you have a loaded image of at least 1px width and 1px wide.

Reply:
Got it.